### PR TITLE
Kill correctly Besu node processes

### PR DESCRIPTION
**Pull request by hmijail**
Close leaked clusters
Log a corner case

### CODE REVIEW

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ProcessBesuNodeRunner.java [hunk 1]
```diff
@@ -361,17 +360,25 @@ public boolean isActive(final String nodeName) {
...
-              if (process.isAlive()) {
-                process.destroy();
-                besuProcesses.remove(name);
-                return false;
-              } else {
-                besuProcesses.remove(name);
-                return true;
-              }
-            });
+    Process p = besuProcesses.remove(name);
```
Comment (CjHare) [inline acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ProcessBesuNodeRunner.java]: final modifier
```suggestion
    final Process p = besuProcesses.remove(name);
```

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ProcessBesuNodeRunner.java [hunk 2]
```diff
@@ -361,17 +360,25 @@ public boolean isActive(final String nodeName) {
...
+      LOG.error("Process {} wasn't in our list", name);
+    }
+
+    process.destroy();
+    try {
+      process.waitFor(2, TimeUnit.SECONDS);
+    } catch (InterruptedException e) {
+      LOG.warn("Wait for death of process " + name + " was interrupted", e);
+    }
+    if (process.isAlive()) {
```
Comment (CjHare) [inline acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ProcessBesuNodeRunner.java]: Consistent new line white space
```suggestion

    if (process.isAlive()) {
```

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ProcessBesuNodeRunner.java [hunk 3]
```diff
@@ -361,17 +360,25 @@ public boolean isActive(final String nodeName) {
...
+    Process p = besuProcesses.remove(name);
+    if (p == null) {
+      LOG.error("Process {} wasn't in our list", name);
+    }
+
+    process.destroy();
+    try {
+      process.waitFor(2, TimeUnit.SECONDS);
+    } catch (InterruptedException e) {
+      LOG.warn("Wait for death of process " + name + " was interrupted", e);
```
Comment (CjHare) [inline acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ProcessBesuNodeRunner.java]: Consistency with other log messages
```suggestion
      LOG.warn("Wait for death of process "{} was interrupted", name, e);
```

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ProcessBesuNodeRunner.java [hunk 4]
```diff
@@ -361,17 +360,25 @@ public boolean isActive(final String nodeName) {
...
+    }
+    if (process.isAlive()) {
+      LOG.warn("Process {} still alive, destroying forcibly now", name);
+      try {
+        process.destroyForcibly().waitFor(2, TimeUnit.SECONDS);
+      } catch (Exception e) {
```
Comment (CjHare) [inline acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ProcessBesuNodeRunner.java]: `final` modifier
```suggestion
      } catch (final Exception e) {
```

File: acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ProcessBesuNodeRunner.java [hunk 5]
```diff
@@ -361,17 +360,25 @@ public boolean isActive(final String nodeName) {
...
-            });
+    Process p = besuProcesses.remove(name);
+    if (p == null) {
+      LOG.error("Process {} wasn't in our list", name);
+    }
+
+    process.destroy();
+    try {
+      process.waitFor(2, TimeUnit.SECONDS);
+    } catch (InterruptedException e) {
```
Comment (CjHare) [inline acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/ProcessBesuNodeRunner.java]: `final` modifier 
```suggestion
    } catch (final InterruptedException e) {
```
